packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,251 @@
+    /// @dev MUST revert if there is a challenge for the safe
+    /// @dev MUST set challenge_start_time to the current block time
+    /// @dev MUST emit the ChallengeStarted event
+    /// @param _safe The Safe to challenge
+    function startChallenge(address _safe) external {
+        SafeConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        if (msg.sender != config.fallbackOwner) {
+            revert LivenessModule2_UnauthorizedCaller();
+        }
+
+        if (config.challengeStartTime != 0) {
+            revert LivenessModule2_ChallengeAlreadyExists();
+        }
+
+        config.challengeStartTime = block.timestamp;
+        emit ChallengeStarted(_safe, block.timestamp);
+    }
+
+    /// @notice Cancels a challenge for an enabled safe
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST revert if there isn't a challenge for the calling safe
+    /// @dev MUST revert if there is a challenge for the calling safe but the challenge is successful
+    /// @dev MUST emit the ChallengeCancelled event
+    function cancelChallenge() external {
+        SafeConfig storage config = safeConfigs[msg.sender];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        if (config.challengeStartTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if challenge period has expired
+        if (block.timestamp >= config.challengeStartTime + config.livenessChallengePeriod) {
+            revert LivenessModule2_ChallengeNotSuccessful();
+        }
+
+        config.challengeStartTime = 0;
+        emit ChallengeCancelled(msg.sender);
+    }
+
+    /// @notice With a successful challenge, removes all current owners from an enabled safe, appoints fallback as its
+    /// sole owner, and sets its quorum to 1
+    /// @dev MUST be executable by anyone
+    /// @dev MUST revert if the given safe hasn't enabled the module
+    /// @dev MUST revert if there isn't a successful challenge for the given safe
+    /// @dev MUST enable the module to start a new challenge
+    /// @dev MUST emit the ChallengeExecuted event
+    /// @param _safe The Safe to transfer ownership of
+    function changeOwnershipToFallback(address _safe) external {
+        SafeConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        if (config.challengeStartTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if challenge period has expired
+        if (block.timestamp < config.challengeStartTime + config.livenessChallengePeriod) {
+            revert LivenessModule2_ChallengeNotSuccessful();
```

Comment:
This should be a different error message, either `LivenessModule2_ChallengeNotFinished` or `LivenessModule2_ResponsePeriodNotExpired` if my renaming above is accepted.

Reply:
Fixed on https://github.com/ethereum-optimism/optimism/pull/17272/commits/989ae745c482be07f1377bcaa220a51fc70b4378